### PR TITLE
generate ID Number together with birthday and gender

### PR DESCRIPTION
**Pull request by asolntsev**
this commit adds method `IdNumber.valid(IdNumberRequest)` which returns a valid combination of ID number, birthday and gender.

### Usage:
```java
        PersonIdNumber person = faker.idNumber().valid(new IdNumberRequest(18, 96, ANY));
        PersonIdNumber person = faker.idNumber().valid(new IdNumberRequest(25, 45, FEMALE));
```

### Problem to solve:
In many countries, ID Number contains or depends on birthday and/or gender.
We need to generate a valid person containing a valid combination of ID number, birthday and gender matching each other. 

Until now, it was possible to generate them separately, but they didn't always match each other:

```java
        String idNumber = faker.idNumber().valid();  // contains birthday of 80 old man
        LocalDate birthday = faker.timeAndDate().birthday();  // birthday of 16 old woman
```

### CODE REVIEW

**Comment by bodiam** [comment]
That looks good to me. @kingthorin , any concerns / feedback?

**Comment by kingthorin** [comment]
I'll try to review this in the next few hours.

File: src/main/java/net/datafaker/idnumbers/MacedonianIdNumber.java [hunk 1]
```diff
@@ -64,8 +74,8 @@ private String rr(BaseProviders faker) {
      * - from 000 to 499 for the male, and
      * - from 500 to 999 for the female citizens.
      */
-    private String sss(BaseProviders faker, boolean female) {
-        int ordinal = female ? faker.number().numberBetween(500, 1000) : faker.number().numberBetween(0, 500);
+    private String sss(BaseProviders faker, Gender gender) {
+        int ordinal = gender == FEMALE ? faker.number().numberBetween(500, 1000) : faker.number().numberBetween(0, 500);
```
Comment (kingthorin) [inline src/main/java/net/datafaker/idnumbers/MacedonianIdNumber.java:78]: I know it was like this before but is 500 really valid for both?

**Review by kingthorin — COMMENTED** [review]
Only see the one item.

I also predict that we'll have to allow other genders or "prefer not to say" in the future, but I guess that doesn't matter until countries start publishing new specs.

**Comment by kingthorin** [comment]
My point was more that it should probably be
0-499
500-1000
Or
0-500
501-1000

If it's supposed to be gender specific how can 500 be valid in both cases?

**Comment by asolntsev** [comment]
> we'll have to allow other genders

Here we are talking only about "genders used for ID numbers". As far as I know, all countries either don't use gender at all, or use MALE/FEMALE option.

> If it's supposed to be gender specific how can 500 be valid in both cases?

Yes, this is a valid code because `numberBetween` function has `(minInclusive, maxExclusive)` parameters. 
In other words, 
*`numberBetween(500, 1000)` means `500 <= X < 1000`
*`numberBetween(1, 500)` means `1 <= X < 500`.

**Comment by kingthorin** [comment]
Okay sorry, thought it was inclusive

**Review by kingthorin — APPROVED** [review]

**Comment by bodiam** [comment]
@asolntsev want to merge this one?

**Comment by asolntsev** [comment]
Sure, yes. :)

**Pull request merged**